subprojects/language-java/src/integTest/groovy/org/gradle/language/java/JavaLanguageIncrementalBuildIntegrationTest.groovy

```diff
@@ -38,4 +38,33 @@ class JavaLanguageIncrementalBuildIntegrationTest extends AbstractJvmLanguageInc
         executedAndNotSkipped ":compileMainJarMainJava", ":createMainJar", ":mainJar"
     }
 
-}
\ No newline at end of file
+    def "task outcome is up to date when no recompilation necessary"() {
+        given:
+        buildFile << '''
+            tasks.withType(org.gradle.api.tasks.compile.AbstractCompile) { task ->
+                task.options.incremental = true
+                task.inputs.file 'input.txt' 
```

Comment:
I think it would be better to change a resource file instead of adding a input which is not on the classpath at all. I do not know if this is true for a single project build but it should be easily possible for a multi project build, right?
Doing it via `inputs` seems somewhat artificial and does not represent the intended use case IMO.